src/DataValidator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace League\ISO3166;
+
+class DataValidator
+{
+    use KeyValidators;
+
+    /**
+     * @param array $data
+     *
+     * @return array
+     */
+    public function validate(array $data)
+    {
+        foreach ($data as $entry) {
+            $this->assertEntryHasRequiredKeys($entry);
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param array $entry
+     *
+     * @throws \DomainException if given data entry does not have all the required keys.
+     */
+    private function assertEntryHasRequiredKeys(array $entry)
```

Comment:
It’s not a problem for now, it’s a problem for later. What ever is public is binding. Limiting what you expose allows you to have more flexibility in the long run and removes some of the burden of maintaining BC.

> On 04 Jul 2016, at 20:31, ignace nyamagana butera <notifications@github.com> wrote:
> 
> In src/DataValidator.php <https://github.com/thephpleague/iso3166/pull/18#discussion_r69484588>:
> 
> > +     */
> > +    public function validate(array $data)
> > +    {
> > +        foreach ($data as $entry) {
> > +            $this->assertEntryHasRequiredKeys($entry);
> > +        }
> > +
> > +        return $data;
> > +    }
> > +
> > +    /**
> > +     * @param array $entry
> > +     *
> > +     * @throws \DomainException if given data entry does not have all the required keys.
> > +     */
> > +    private function assertEntryHasRequiredKeys(array $entry)
> @frankdejonge <https://github.com/frankdejonge> Since this is a new class I don't see the problem regarding public API ?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/thephpleague/iso3166/pull/18/files/7769901aea7b2ed28c11d9ee1ecc0566fab9df44#r69484588>, or mute the thread <https://github.com/notifications/unsubscribe/AAgopWYUHqp7atQF5o1tMTCYg70IBrZfks5qSVGQgaJpZM4JDIDq>.
>